projects/storefrontlib/src/events/product/product-page-event.builder.ts

```diff
@@ -72,23 +72,20 @@ export class ProductPageEventBuilder {
       skip(1)
     );
 
-    const categoryPageEvent$ = this.eventService.get(PageEvent).pipe(
-      map((pageEvent) => ({
-        isCategoryPage: pageEvent.semanticRoute === 'category',
-        categoryCode: pageEvent.context.id,
-      }))
-    );
-
-    return categoryPageEvent$.pipe(
+    return this.eventService.get(PageEvent).pipe(
       switchMap((pageEvent) => {
-        if (!pageEvent.isCategoryPage) {
+        if (pageEvent.semanticRoute !== 'category') {
           return EMPTY;
         }
 
         return searchResults$.pipe(
           map((searchResults) => ({
-            categoryCode: pageEvent.categoryCode,
-            categoryName: searchResults.breadcrumbs[0].facetValueName,
+            ...pageEvent,
+            categoryCode: pageEvent.context.id,
```

Comment:
what about:

```typescript
...{
  categoryCode: pageEvent?.context?.id,
  categoryName: searchResults.breadcrumbs?.[0].facetValueName,
  numberOfResults: searchResults.pagination?.totalResults,
}
```

as other values might be not available based on the occ param config.

Reply:
Done